cmake/modules/build_configuration.cmake

```diff
@@ -0,0 +1,107 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2021, Nordic Semiconductor ASA
+
+# Configure Zephyr build system.
+#
+# This CMake module will setup various configuration settings based on user input.
+#
+# Outcome:
+# The following variables will be defined when this CMake module completes:
+#
+# - CONF_FILE:               List of Kconfig fragments
+# - DTC_OVERLAY_FILE:        List of devicetree overlay files
+# - APPLICATIUON_CONFIG_DIR: Root folder for application configuration
+#
+# If any of the above variables are already set when this CMake module is
+# loaded, then no changes to the variable will happen.
+#
+# CMake module dependencies:
+# - extension CMake module.
```

Comment:
In which way ?
It is correct that the `BOARD` variable must be set (which should be mentioned in the description, so will add).

But the `board.cmake` itself is not be a direct dependency.
One may reuse this module without re-using the `board.cmake` for example to lookup just configuration files without a board validation.

Reply:
Why would one want to opt-out of board vaildation when looking up configuration files?

Reply:
> Why would one want to opt-out of board vaildation when looking up configuration files?

In case you want to re-use the kconfig infrastructure for other purposes.

But the main point here is not to support such use-cases, but more the fact to have clear split as to when a module has a hard dependency on another module and when it's a looser coupling as described here: https://github.com/zephyrproject-rtos/zephyr/pull/41301#discussion_r777444109

Following that description, it makes it clearer when two modules are considered to have hard dependencies, and when they are just loosely coupled.

Reply:
Since this is just for internal use we can know of all that include build_configuration.cmake and ensure that boards.cmake is included before this is included when appropriate.

Reply:
although internal use is correct, then internal use is also reusing in different contexts which might have different requirements, such as #40555.

#40555 is also internal use, but doesn't want to use all the same modules as a regular Zephyr build does.